src/components/Web3Modal.tsx

```diff
@@ -1,127 +1,174 @@
-import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import {
   StyleSheet,
   View,
   useColorScheme,
   ImageBackground,
+  Alert,
 } from 'react-native';
 import Modal from 'react-native-modal';
 
 import { DEVICE_WIDTH } from '../constants/Platform';
 import { DarkTheme, LightTheme } from '../constants/Colors';
-import type { Routes } from '../constants/Routes';
-import type { Listing } from '../types/controllerTypes';
 
 import Background from '../assets/Background.png';
 import Web3ModalHeader from './Web3ModalHeader';
-import { ExplorerUtil } from '../utils/ExplorerUtil';
-import InitialExplorer from '../views/InitialExplorer';
-import ViewAllExplorer from '../views/ViewAllExplorer';
-import QRCodeView from '../views/QRCodeView';
-
-const INITIAL_ROUTE = 'INIT_WALLETS';
+import { createUniversalProvider, createSession } from '../utils/ProviderUtil';
+import { ModalCtrl } from '../controllers/ModalCtrl';
+import { Web3ModalRouter } from './Web3ModalRouter';
+import { ExplorerCtrl } from '../controllers/ExplorerCtrl';
+import { ConfigCtrl } from '../controllers/ConfigCtrl';
+import { OptionsCtrl } from '../controllers/OptionsCtrl';
+import { ClientCtrl } from '../controllers/ClientCtrl';
 
 interface Web3ModalProps {
   projectId: string;
-  isVisible: boolean;
-  onClose: () => void;
-  currentWCURI?: string;
+  relayUrl: string;
+  onCopyClipboard?: (value: string) => void;
 }
 
-export function Web3Modal({ projectId, isVisible, onClose }: Web3ModalProps) {
-  const [isWalletListLoading, setWalletListLoading] = useState(true);
-  const [initialWallets, setInitialWallets] = useState<Listing[]>([]);
-  const [allWallets, setAllWallets] = useState<Listing[]>([]);
+export function Web3Modal({
+  projectId,
+  relayUrl,
+  onCopyClipboard,
+}: Web3ModalProps) {
+  const [modalVisible, setModalVisible] = useState(false);
 
   const isDarkMode = useColorScheme() === 'dark';
 
-  const [viewStack, setViewStack] = useState<Routes[]>([INITIAL_ROUTE]);
+  const onSessionCreated = useCallback(async () => {
+    OptionsCtrl.getAccount();
+    ModalCtrl.close();
+  }, []);
 
-  const fetchWallets = useCallback(() => {
-    ExplorerUtil.fetchWallets(projectId).then((wallets) => {
-      setWalletListLoading(false);
-      if (wallets) {
-        setInitialWallets(wallets.listings.slice(0, 7));
-        setAllWallets(wallets.listings);
-      }
-    });
-  }, [projectId]);
+  const onSessionError = useCallback(async () => {
+    // TODO: Improve this, check why is alerting a lot, and check MaxListeners warning
```

Comment:
For context, `MaxListeners` is likely coming from sign sdk (we saw this there as well), unless it's something else.